pandas/pandas_interoperability

```diff
@@ -0,0 +1,27 @@
+import Orange
+import pandas as pd
+import numpy as np
+
+
+def construct_domain(df):
+    columns = OrderedDict(df.dtypes)
+
+    def create_variable(col):
+        if col[1].__str__().startswith('float'):
```

Comment:
Hi Kern,

I am having an error when using many widgets, I just want to know if this
is could be produced by the way I am converting dataframes between pandas
and Orange.

Best,
Jose

TypeError Traceback (most recent call last):

File "c:\users\jamh\orange3\orange3\Orange\canvas\scheme\widgetsscheme.py",
line 680, in process_signals_for_widget

handler(*args)

File "c:\users\jamh\orange3\orange3\Orange\widgets\utils\sql.py", line 24,
in new_f

return f(self, data, *args, **kwargs)

File "c:\users\jamh\orange3\orange3\Orange\widgets\data\owrank.py", line
256, in setData

self.updateScores()

File "c:\users\jamh\orange3\orange3\Orange\widgets\data\owrank.py", line
313, in updateScores

self.measure_scores[index] = estimator(data)

File "c:\users\jamh\orange3\orange3\Orange\preprocess\score.py", line 48,
in __call__

data = pp(data)

File "c:\users\jamh\orange3\orange3\Orange\preprocess\preprocess.py", line
110, in __call__

attributes = [transform(var) for var in data.domain.attributes]

File "c:\users\jamh\orange3\orange3\Orange\preprocess\preprocess.py", line
110, in <listcomp>

attributes = [transform(var) for var in data.domain.attributes]

File "c:\users\jamh\orange3\orange3\Orange\preprocess\preprocess.py", line
100, in transform

new_var = method(data, var)

File "c:\users\jamh\orange3\orange3\Orange\preprocess\discretize.py", line
121, in __call__

d = distribution.get_distribution(data, attribute)

File "c:\users\jamh\orange3\orange3\Orange\statistics\distribution.py",
line 296, in get_distribution

return Continuous(dat, variable, unknowns)

File "c:\users\jamh\orange3\orange3\Orange\statistics\distribution.py",
line 194, in __new__

return cls.from_data(variable, dat)

File "c:\users\jamh\orange3\orange3\Orange\statistics\distribution.py",
line 213, in from_data

dist, unknowns = data._compute_distributions([variable])[0]

File "c:\users\jamh\orange3\orange3\Orange\data\table.py", line 1308, in
_compute_distributions

unknowns = bn.countnans(m)

File "countnans.pyx", line 107, in countnans.countnans
(bottlechest/src/func/32bit/countnans.c:1788)

File "countnans.pyx", line 698, in countnans.countnans_slow_axisNone
(bottlechest/src/func/32bit/countnans.c:12306)

File "C:\Anaconda3\lib\site-packages\bottlechest\slow\func.py", line 261,
in countnans

return np.sum(np.isnan(arr))

TypeError: ufunc 'isnan' not supported for the input types, and the inputs
could not be safely coerced to any supported types according to the casting
rule ''safe''



2015-10-26 11:05 GMT+01:00 kernc <notifications@github.com>:

> In pandas/pandas_interoperability
> <https://github.com/biolab/orange3/pull/796#discussion_r42975152>:
>
> > @@ -0,0 +1,27 @@
> > +import Orange
> > +import pandas as pd
> > +import numpy as np
> > +
> > +
> > +def construct_domain(df):
> > +    columns = OrderedDict(df.dtypes)
> > +
> > +    def create_variable(col):
> > +        if col[1].__str__().startswith('float'):
>
> Apparently, one can use __eq__() method to test for dtype equality, e.g.
>
> if dtype == float: ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biolab/orange3/pull/796/files#r42975152>.
>



-- 
/ .- .-.. .-.. / -.-- --- ..- / -. . . -.. / .. ... / .-.. --- ...- .
José Antonio Martín H. (PhD)           E-Mail: jamartinh@fdi.ucm.es
Computer Science Faculty               Phone: (+34) 91 3947650
Complutense University of Madrid       Fax: (+34) 91 3947527
C/ Prof. José García Santesmases,s/n   28040 Madrid, Spain
web: http://www.dacya.ucm.es/jam/
LinkedIn: http://www.linkedin.com/in/jamartinh (Let's connect)
.-.. --- ...- . / .. ... / .- .-.. .-.. / .-- . / -. . . -..